cb-check/cb_check.ml

```diff
@@ -0,0 +1,52 @@
+let read_all ic =
+  let b = Buffer.create 80 in
+  let rec loop () =
+    let line = input_line ic in
+    Buffer.add_string b (line ^ "\n");
+    loop ()
+  in
+  try loop ()
+  with End_of_file ->
+    close_in ic;
+    Buffer.contents b
+
+let validate l =
+  let rec loop l acc =
+    match l with
+    | [] -> Some acc
+    | (str, (_, _)) :: l -> (
+        match Yojson.Safe.from_string str with
+        | j -> (
+            match Cb_schema.S.of_json j with
+            | s -> loop l (s :: acc)
+            | exception Invalid_argument s ->
+                Format.eprintf
+                  "Some valid json didn't conform to the schema with error: %S\n\
+                   The json: %a\n"
+                  s
+                  (Yojson.Safe.pretty_print ~std:false)
+                  j;
+                None)
+        | exception Yojson.Json_error s ->
+            Format.eprintf
+              "\x1b[91mJson parsing failure, please report: \x1b[0m%s" s;
+            None)
+  in
+  match loop l [] with
```

Comment:
Perhaps the `loop` can be replaced by a `List.filter_map`? (It's not technically a blocking error if one of the detected json fails, but it is nice to report it as a warning! The real error is if no schema-valid jsons are detected at all by the end)

Reply:
If I understood correctly, this should be fixed now

Reply:
Yes it looks good! Thanks :)